src/Modules/Data/Commands.Data.Test/GetPowerBITableTests.cs

```diff
@@ -0,0 +1,109 @@
+                var dataSetResult = ps.Invoke();
+                ps.Commands.Clear();
+                ps.AddCommand(GetPowerBITableCmdletInfo)
+                    .AddParameter("DatasetId", dataSetResult.Where(x => (bool)x.Members["addRowsAPIEnabled"].Value == true).FirstOrDefault().Members["Id"].Value)
+                    .AddParameter("Name", "Product");
+                
+                // Act
+                var result = ps.Invoke();
+
+                // Assert
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(result);
+                Assert.IsTrue(result.Count > 0);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("Interactive")]
+        [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
+        public void EndToEndGetPowerBITable_DatasetParameterSetName()
+        {
+            using (var ps = System.Management.Automation.PowerShell.Create())
+            {
+                // Arrange
+                ProfileTestUtilities.ConnectToPowerBI(ps, "Public");
+                ps.AddCommand(GetPowerBIDatasetCmdletInfo).AddParameter("Scope", PowerBIUserScope.Individual);
+                ps.AddCommand("Where-Object");
+                var filter = ScriptBlock.Create("$_.AddRowsApiEnabled -eq $true");
+                ps.AddParameter("FilterScript", filter);
+                ps.AddCommand(GetPowerBITableCmdletInfo);
+
+                // Act
+                var result = ps.Invoke();
+                // Assert
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(result);
+                Assert.IsTrue(result.Count > 0);
+            }
+        }
+
+        [TestMethod]
+        public void GetPowerBITable_DatasetIdParameterSetName()
+        {
+            var expectedDatasets = new List<Dataset> { new Dataset { Id = Guid.NewGuid(), Name = "TestDataset" } };
+            var expectedTables = new List<Table> { new Table {  Name = "TestTable" } };
+            var client = new Mock<IPowerBIApiClient>();
+            client.Setup(x => x.Datasets.GetDatasets()).Returns(expectedDatasets);
+            var datasetInitFactory = new TestPowerBICmdletInitFactory(client.Object);
+            var getDatasetCmdlet = new GetPowerBIDataset(datasetInitFactory);
+            getDatasetCmdlet.InvokePowerBICmdlet();
+            var datasets = datasetInitFactory.Logger.Output.ToList();
+            var tableInitFactory = new TestPowerBICmdletInitFactory(client.Object);
+            client.Setup(x => x.Datasets.GetTables(((Dataset)datasets[0]).Id, null)).Returns(expectedTables);
+            var getTableCmdlet = new GetPowerBITable(tableInitFactory);
+            getTableCmdlet.DatasetId = ((Dataset)datasets[0]).Id;
+
+            // Act
+            getTableCmdlet.InvokePowerBICmdlet();
+
+            // Assert
+            AssertExpectedUnitTestResults(expectedTables, tableInitFactory);
```

Comment:
You can actually replace this with `tableInitFactory.AssertExpectedUnitTestResults(expectedTables)`
See GetPowerBIDatasourceTests.cs for examples.

Reply:
I did.